core/commands/add.go

```diff
@@ -277,16 +318,24 @@ func addDir(n *core.IpfsNode, dir files.File, out chan interface{}, progress boo
 			break
 		}
 
-		node, err := addFile(n, file, out, progress, false)
-		if err != nil {
+		node, err := addFile(n, file, out, progress, false, hidden, localIgnorePatterns)
+		if _, ok := err.(*hiddenFileError); ok {
+			// hidden file error, set the node to nil for below
+			node = nil
+		} else if _, ok := err.(*ignoreFileError); ok {
+			// ignore file error, set the node to nil for below
+			node = nil
+		} else if err != nil {
```

Comment:
(it may be a matter of counting the files added, and in the root function check:

```go
// when adding or ignoring, incr counters:
adr.filesAdded++
adr.filesIgnored++

func (adr *adder) ignoredAllFiles() bool {
  return adr.filesAdded == 0 && adr.filesIgnored > 0
}

if adder.IgnoredAllFiles {
  return errIgnoredAllFiles
}
```